client/state/analytics/actions.js

```diff
@@ -0,0 +1,47 @@
+import fromPairs from 'lodash/fromPairs';
+
+import {
+	BUMP_STAT,
+	RECORD_EVENT,
+	RECORD_PAGE_VIEW
+} from './constants.js';
+
+export const withAnalytics = ( analytics, action ) =>
```

Comment:
Actually, I realized that we could trivially support thunks by simply sequentially running the actions via a new thunk. This doesn't actually help us too much in terms of pushing us towards pure action creators, but it seems safe and simple for a transition.

In other words, thunk-in-thunk-out.